src/org/zaproxy/zap/extension/httppanel/MessageChangedListener.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Zed Attack Proxy (ZAP) and its related class files.
+ *
+ *  ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ *  Copyright 2018 The ZAP Development Team
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.httppanel;
+
+public interface MessageChangedListener {
```

Comment:
- I'd suggest renaming it to `MessageDisplayedListener`(?), to avoid any misunderstandings with the editable panels (where the messages themselves can be changed).
 - Worth adding a short description of the listener (and `@since` tag), it should also link to `HttpPanel`'s method that allows to add the listener and mention the `ExtensionHookView`.

Reply:
Is
```suggestion
public interface DisplayedMessageChangedListener {
```
okay for you?

Reply:
Yeah, that's clear too.